spec.md

```diff
@@ -100,6 +110,18 @@ interface ExpressionStatement <: Statement {
 
 An expression statement, i.e., a statement consisting of a single expression.
 
+## Directive
+
+```js
+interface Directive <: ExpressionStatement {
+    directive: string;
+}
+```
+
+A directive from the directive prologue of a script or function.
+The `directive` property is the raw string source of the directive.
+This source includes single or double wrapping quotes and may not contain an EscapeSequence or LineContinuation..
```

Comment:
> I see your point about `use\x20strict` being valid directive, but your other examples can be totally covered by boolean `directive` and I don't really see arguments why you claim it can't be.

Because the presence of escape sequences and line continuations—which are not exposed in ESTree—is relevant. String `ExpressionStatement.directive` and `ExpressionStatement.expression.value` are _not_ duplicate data; but they are obviously related (and coincide in common cases).

> having yet another CST property (`directive` string) in tree, when we have `value` and semi-standardized `raw` already.

I disagree that string `directive` is not abstract (i.e., belonging in strictly optional CST properties), because the distinctions matter for directives in a way that they don't for non-directive literals. And you can't point to `raw`, not just because it isn't part of the spec but also because _all_ properties can be derived from raw source.